### PR TITLE
Use own validator ClassUtils.getName to get String input

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

**Comment by garydgregory** [comment]
@arturobernalg Would you rebase on master, please?

**Comment by arturobernalg** [comment]
> @arturobernalg Would you rebase on master, please?

Done
TY

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 1]
```diff
@@ -3566,7 +3566,7 @@ public static boolean isSameType(final Object array1, final Object array2) {
         if (array1 == null || array2 == null) {
             throw new IllegalArgumentException("The Array must not be null");
         }
-        return array1.getClass().getName().equals(array2.getClass().getName());
+        return ClassUtils.getName(array1).equals(ClassUtils.getName(array2));
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java:3569]: Why? See my comments in your other PR. In this case, neither array1 or array2 can be null here.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Yep, my bad. changed.

File: src/main/java/org/apache/commons/lang3/time/FastDatePrinter.java [hunk 2]
```diff
@@ -414,7 +415,7 @@ public StringBuffer format(final Object obj, final StringBuffer toAppendTo, fina
             return format(((Long) obj).longValue(), toAppendTo);
         }
         throw new IllegalArgumentException("Unknown class: " +
-            (obj == null ? "<null>" : obj.getClass().getName()));
+            (ClassUtils.getName(obj,  "<null>")));
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/time/FastDatePrinter.java:418]: Don't need extra parentheses or extra space after comma.

File: src/test/java/org/apache/commons/lang3/EnumUtilsTest.java [hunk 3]
```diff
@@ -378,7 +378,7 @@ public void test_getEnumMap_keyFunction() {
 
     @Test
     public void test_getEnumSystemProperty() {
-        final String key = getClass().getName();
+        final String key = ClassUtils.getName(this);
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/EnumUtilsTest.java:381]: Why?

File: src/test/java/org/apache/commons/lang3/reflect/TypeUtilsTest.java [hunk 4]
```diff
@@ -940,7 +941,7 @@ public void testParameterizeWithOwner() throws Exception {
 
     @Test
     public void testToLongString() {
-        assertEquals(getClass().getName() + ":B", TypeUtils.toLongString(getClass().getTypeParameters()[0]));
+        assertEquals(ClassUtils.getName(this) + ":B", TypeUtils.toLongString(getClass().getTypeParameters()[0]));
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/reflect/TypeUtilsTest.java:944]: Why?

**Review by garydgregory — CHANGES_REQUESTED** [review]
@arturobernalg 
Please see my comments.

File: src/test/java/org/apache/commons/lang3/EnumUtilsTest.java (hunk 3 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/EnumUtilsTest.java]: changed.

File: src/test/java/org/apache/commons/lang3/reflect/TypeUtilsTest.java (hunk 4 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/reflect/TypeUtilsTest.java]: changed.

**Comment by garydgregory** [comment]
@arturobernalg 
The PR fails. Just run "mvn" locally before pushing to run the default goals.

**Comment by arturobernalg** [comment]
> @arturobernalg
> The PR fails. Just run "mvn" locally before pushing to run the default goals.

Solved.

**Pull request closed without merging**